src/integrations/pathManager.ts

```diff
@@ -0,0 +1,108 @@
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import paths from '@/utils/paths';
+import { manageLinesInFile } from '@/integrations/manageLinesInFile';
+
+let pathManager: ManualPathManager;
+
+// PathManager is the interface that anything that manages the
+// PATH variable must implement.
+interface PathManager {
+  // Makes real any changes to the system. Should be idempotent.
+  enforce(): Promise<void>
+  // Removes any changes that the PathManager may have made.
+  // Should be idempotent.
+  remove(): Promise<void>
+}
+
+// ManualPathManager is for when the user has chosen to manage
+// their PATH themselves. It does nothing.
+class ManualPathManager implements PathManager {
+  async enforce(): Promise<void> {}
+  async remove(): Promise<void> {}
+}
+
+// RcFilePathManager is for when the user wants Rancher Desktop to
+// make changes to their PATH by putting lines that change it in their
+// shell .rc files.
+export class RcFilePathManager implements PathManager {
+  constructor() {
+    const platform = os.platform();
+
+    if (platform !== 'linux' && platform !== 'darwin') {
+      throw new Error(`Platform "${ platform }" is not supported by RcFilePathManager`);
+    }
+  }
+
+  async enforce(): Promise<void> {
+    await this.managePosix(true);
+    await this.manageCsh(true);
+    await this.manageFish(true);
+  }
+
+  async remove(): Promise<void> {
+    await this.managePosix(false);
+    await this.manageCsh(false);
+    await this.manageFish(false);
+  }
+
+  protected async managePosix(desiredPresent: boolean): Promise<void> {
+    const pathLine = `PATH=\${PATH}:${ paths.integration }`;
+
+    await Promise.all(['.bashrc', '.zshrc'].map((rcName) => {
+      const rcPath = path.join(os.homedir(), rcName);
+
+      return manageLinesInFile(rcPath, [pathLine], desiredPresent);
+    }));
+  }
+
+  protected async manageCsh(desiredPresent: boolean): Promise<void> {
+    const pathLine = `set path=($path ${ paths.integration })`;
+
+    await Promise.all(['.cshrc', '.tcshrc'].map((rcName) => {
+      const rcPath = path.join(os.homedir(), rcName);
+
+      return manageLinesInFile(rcPath, [pathLine], desiredPresent);
+    }));
+  }
+
+  protected async manageFish(desiredPresent: boolean): Promise<void> {
+    const pathLine = `set -x PATH "${ paths.integration }" "$PATH"`;
+    let configHome = '';
+
+    switch (os.platform()) {
+    case 'darwin':
+      configHome = path.join(os.homedir(), '.config');
+      break;
+    case 'linux':
+      configHome = process.env['XDG_CONFIG_HOME'] || path.join(os.homedir(), '.config');
+      break;
```

Comment:
We don't need to do the OS check here, do we? `XDG_CONFIG_HOME` would just never be set on mac.

Reply:
Right you are, fixed.